src/fsharp/pickle.fs

```diff
@@ -1411,15 +1411,41 @@ let u_trait st =
 #if INCLUDE_METADATA_WRITER
 let p_rational q st = p_int32 (GetNumerator q) st; p_int32 (GetDenominator q) st
 
-let rec p_measure_expr unt st =
-    let unt = stripUnitEqnsAux false unt 
-    match unt with 
-    | MeasureCon tcref   -> p_byte 0 st; p_tcref "measure" tcref st
-    | MeasureInv x       -> p_byte 1 st; p_measure_expr x st
-    | MeasureProd(x1,x2) -> p_byte 2 st; p_measure_expr x1 st; p_measure_expr x2 st
-    | MeasureVar(v)      -> p_byte 3 st; p_tpref v st
-    | MeasureOne         -> p_byte 4 st
-    | MeasureRationalPower(x,q) -> p_byte 5 st; p_measure_expr x st; p_rational q st
+let p_measurecon tcref st = p_byte 0 st; p_tcref "measure" tcref st
```

Comment:
This is all sublte enough and irregular enough that it wouldd be good to add a comment explaining that we are encoding as much as possible into the non-rational-unit subset. Add a link to this bug as well (I expect we are adopting a policy where adding links to GitHub links into code is acceptable)